src/backend/distributed/executor/local_executor.c

```diff
@@ -506,3 +507,35 @@ DisableLocalExecution(void)
 					  (superuser() ? PGC_SUSET : PGC_USERSET), PGC_S_SESSION,
 					  GUC_ACTION_LOCAL, true, 0, false);
 }
+
+
+/*
+ * TaskQueryString generates task->queryStringLazy if missing.
+ *
+ * For performance reasons, the queryString is generated lazily. For example
+ * for local queries it is always needed to generate it.
```

Comment:
Minor: `For example
 * for local queries it is always needed to generate it.` this part of the comment sounds a little confusing to me. For some  of the local queries, we skip generating the query string?